lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/fs.rb

```diff
@@ -164,30 +178,59 @@ def cmd_search(*args)
           globs << val
         when "-r"
           recurse = false if val =~ /^(f|n|0)/i
+        when "-a"
+          startDate = val
+        when "-b"
+          endDate = val
       end
     }
+    if startDate.to_s.empty?
+        startDate = ''
+    else
+        if !vali_date(startDate)
+            print_error("Bad time specification or invalid time.  You must use this datetime format: \"YYYY-mm-ddTHH:MM:SS\" , e.g. >search -a \"2000-04-30T16:58:31\" ")
+            return
+        end
+    end
 
+    if endDate.to_s.empty?
+        endDate = ''
+    else
+        if !vali_date(endDate)
+            print_error("Bad time specification or invalid time.  You must use this datetime format: \"YYYY-mm-ddTHH:MM:SS\" , e.g. >search -b \"2000-04-30T16:58:31\" ")
+            return
+        end
+    end
     if globs.empty?
       print_error("You must specify a valid file glob to search for, e.g. >search -f *.doc")
       return
     end
 
+
     globs.uniq.each do |glob|
-      files += client.fs.file.search(root, glob, recurse)
+      files += client.fs.file.search(root, glob, recurse, startDate, endDate)
     end
 
+
     if files.empty?
       print_line("No files matching your search were found.")
       return
     end
 
     print_line("Found #{files.length} result#{ files.length > 1 ? 's' : '' }...")
+    print_line("%-50s    %10s    %s" % ["FILEPATH","SIZE","MODIFIED (REMOTE UTC)"])
+    print_line("="*80)
```

Comment:
There's a table helper utility in `Rex::Text::Table` that this should use for automatic sizing, wrapping etc.

Reply:
I used table, but was unsure how to use your ColSpec formatters.  I was trying to right-justify filesizes in the table, I will work on that after I get the other payloads done.